internal/publicip/api/cloudflare.go

```diff
@@ -0,0 +1,96 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/netip"
+	"strings"
+
+	"github.com/qdm12/gluetun/internal/constants"
+	"github.com/qdm12/gluetun/internal/models"
+)
+
+type cloudflare struct {
+	client *http.Client
+}
+
+func newCloudflare(client *http.Client) *cloudflare {
+	return &cloudflare{
+		client: client,
+	}
+}
+
+// FetchInfo obtains information on the IP address provided using the
+// Cloudflare API. If the IP is the zero value, the public IP address
+// of the machine is used as the IP. Since the Cloudflare API can only
+// be used to provide details about the public IP, providing a non-zero
+// IP results in an error indicating that the IP is not supported.
```

Comment:
```suggestion
// FetchInfo obtains information on the public IP address of the machine,
// and returns an error if the `ip` argument is set since the Cloudflare API 
// can only be used to provide details about the current machine public IP.
```

Reply:
done